Ah, that's handy, a test failed with the `network is unreachable` error - and the first retry (after 100ms) succeeded. So, at least in this case, it was a timing thing ...

```
=== FAIL: amd64.integration.networking TestDefaultBridgeAddresses/Unique-Local_Subnet_Changes/Modify_UL_prefix_-_no_address_change (0.79s)
    bridge_linux_test.go:642: assertion failed: error is not nil: Error response from daemon: failed to add interface vethe9d524f to sandbox: failed to advertise addresses: write ip ::1->ff02::1: sendmsg: network is unreachable
        success
    --- FAIL: TestDefaultBridgeAddresses/Unique-Local_Subnet_Changes/Modify_UL_prefix_-_no_address_change (0.79s)
```

--- a/integration/networking/bridge_linux_test.go
+++ b/integration/networking/bridge_linux_test.go
@@ -1536,6 +1536,9 @@ func TestAdvertiseAddresses(t *testing.T) {
 
 			icmps := stopICMP6Listen()
 			checkPkts("ICMP6", icmps, netip.MustParseAddr(ctr2Addr6), network.UnpackUnsolNA)
+			if t.Failed() {
+				d.TailLogsT(t, 100)
+			}
 		})
 	}
 }

--- a/libnetwork/osl/interface_linux.go
+++ b/libnetwork/osl/interface_linux.go
@@ -350,6 +350,9 @@ func (n *Namespace) AddInterface(ctx context.Context, srcName, dstPrefix string,
 }
 
 func waitForIfUpped(ctx context.Context, ns netns.NsHandle, ifIndex int) (bool, error) {
+	ctx, span := otel.Tracer("").Start(context.WithoutCancel(ctx), "libnetwork.osl.waitforIfUpped")
+	defer span.End()
+
 	update := make(chan netlink.LinkUpdate, 100)
 	upped := make(chan struct{})
 	opts := netlink.LinkSubscribeOptions{
@@ -398,6 +401,7 @@ func waitForIfUpped(ctx context.Context, ns netns.NsHandle, ifIndex int) (bool, 
 	for {
 		select {
 		case <-timerC:
+			log.G(ctx).Warnf("timeout in waitForIfUpped")
 			return false, nil
 		case u, ok := <-update:
 			if !ok {

--- a/libnetwork/osl/interface_linux.go
+++ b/libnetwork/osl/interface_linux.go
@@ -498,9 +498,37 @@ func (n *Namespace) advertiseAddrs(ctx context.Context, ifIndex int, i *Interfac
 			}
 		}
 		if naSender != nil {
-			if err := naSender.Send(); err != nil {
-				log.G(ctx).WithError(err).Warn("Failed to send unsolicited NA")
-				errs = append(errs, err)
+			// FIXME(robmry) - retry if this fails, but still return the error.
+			// In CI, the send has failed a couple of times with "write ip ::1->ff02::1: sendmsg: network is unreachable".
+			// Can't repro locally, so - try find out whether a retry helps and it's something racing, or it's a
+			// persistent problem.
+			for c := range 3 {
+				if c > 0 {
+					time.Sleep(50 * time.Millisecond)
+				}
+
+				routes, rgErr := nlh.RouteGetWithOptions(net.IPv6linklocalallnodes, &netlink.RouteGetOptions{
+					IifIndex: ifIndex,
+					SrcAddr:  net.IPv6loopback,
+				})
+				var routeStr string
+				if rgErr != nil {
+					routeStr = fmt.Sprintf("RouteGet->'%s'", rgErr.Error())
+				} else if len(routes) != 1 {
+					routeStr = fmt.Sprintf("RouteGet->%d routes", len(routes))
+				} else {
+					routeStr = fmt.Sprintf("RouteGet->'%s'", routes[0].String())
+				}
+
+				if err := naSender.Send(); err != nil {
+					log.G(ctx).WithError(err).Warn("Failed to send unsolicited NA")
+					errs = append(errs, fmt.Errorf("%s: %w", routeStr, err))
+					continue
+				}
+				if c > 0 {
+					errs = append(errs, fmt.Errorf("success"))
+				}
+				break
 			}
 		}
 		return errors.Join(errs...)

